modules/os2web_audit_entity/os2web_audit_entity.module

```diff
@@ -0,0 +1,125 @@
+<?php
+
+/**
+ * @file
+ * Hooks into drupal and collect logging data.
+ */
+
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\Core\Session\AccountInterface;
+use Drupal\os2web_audit_entity\Form\SettingsForm;
+
+/**
+ * Implements hook_entity_insert().
+ */
+function os2web_audit_entity_entity_insert(EntityInterface $entity): void {
+  $msg = sprintf('Entity (%d) of type "%s" created.', $entity->id(), $entity->getEntityTypeId());
+  os2web_audit_entity_log($msg);
+}
+
+/**
+ * Implements hook_entity_update().
+ */
+function os2web_audit_entity_entity_update(EntityInterface $entity): void {
+  $msg = sprintf('Entity (%d) of type "%s" updated.', $entity->id(), $entity->getEntityTypeId());
+  os2web_audit_entity_log($msg);
+}
+
+/**
+ * Implements hook_entity_delete().
+ */
+function os2web_audit_entity_entity_delete(EntityInterface $entity): void {
+  $msg = sprintf('Entity (%d) of type "%s" deleted.', $entity->id(), $entity->getEntityTypeId());
+  os2web_audit_entity_log($msg);
+}
+
+/**
+ * Implements hook_entity_storage_load().
+ *
+ * Logs access for file entities.
+ */
+function os2web_audit_entity_entity_storage_load(mixed $entities, string $entity_type): void {
+  foreach ($entities as $entity) {
+    if ($entity_type == 'file') {
+      /** @var \Drupal\file\Entity\File $entity */
+      $fid = $entity->id();
+      $uri = $entity->getFileUri();
+      $msg = sprintf('File (%d) accessed. Uri "%s"', $fid, $uri);
+      os2web_audit_entity_log($msg);
+    }
+  }
+}
+
+/**
+ * Implements hook_webform_post_load_data().
+ */
+function os2web_audit_entity_webform_post_load_data(mixed $submissions): void {
+  foreach ($submissions as $submission) {
+    // Try to check for _cpr field for extra logging information.
+    $personal = '';
+    $submissionData = $submission->getData();
+    $filterFields = preg_grep('/^(.*)cpr(.*)$/', array_keys($submissionData));
+    if (!empty($filterFields)) {
+      foreach ($filterFields as $field) {
+        $cpr = $submissionData[$field];
+        $personal .= sprintf('CPR "%s".', $cpr);
+      }
+    }
+
```

Comment:
Are we interested in logging submission access if `$personal` is empty?

Reply:
yes, "personal" is just CPR's.... but data-lak of submission with mails, address etc. still nice to known when it happened.